dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/NoOpPermissionsImpl.java

```diff
@@ -43,4 +43,14 @@ public void removePermission(User user, Workflow workflow, String email, Role ro
     public boolean canDoAction(User user, Workflow workflow, Role.Action action) {
         return false;
     }
+
+    @Override
+    public void selfDestruct(User user) {
+        // Do nothing
```

Comment:
throw exceptions here like above for sanity

Reply:
I'm only throwing exceptions for unsupported operations; I don't think I should throw one here.

The idea is that if some worst case scenario happens and we need to turn off the SAM support (I don't know why, just being paranoid), we could make this the active authorizer implementation. This one would not let you set permissions, hence the exceptions, but wouldn't otherwise get in the way, so a user could self destruct.

In looking at this, I did notice that I should have `isSharing` return false, otherwise a user wouldn't be able to self-destruct.